yarn/src/main/scala/org/apache/spark/deploy/yarn/security/ExecutorDelegationTokenUpdater.scala

```diff
@@ -107,6 +110,14 @@ private[spark] class ExecutorDelegationTokenUpdater(
     }
   }
 
+  private def getTimeOfNextUpdateFromFileName(credentialsPath: Path): Long = {
+    val name = credentialsPath.getName
+    val index = name.lastIndexOf(SparkHadoopUtil.SPARK_YARN_CREDS_COUNTER_DELIM)
+    val slice = name.substring(0, index)
+    val last2index = slice.lastIndexOf(SparkHadoopUtil.SPARK_YARN_CREDS_COUNTER_DELIM)
+    name.substring(last2index).toLong
```

Comment:
is this giving you the right time?  It looks like its giving you the credentials file name instead of the next time to update.

name is: credentials-UUID-timestamp-1...   slice = credentials-UUID-timestamp, name.substring of that is credentials-UUID.  don't you want name.substring(last2index, index) ?